lib/api/src/externals/function.rs

```diff
@@ -269,10 +324,25 @@ impl Function {
                 Env::init_with_instance(&mut *env.ctx.env, instance)
             };
 
+        #[ cfg(feature="async") ]
+        let (host_env, metadata) =  {
+            let host_env_set_yielder_fn = |env_ptr, yielder_ptr| {
+                let env: &mut Env = unsafe { std::mem::transmute(env_ptr) };
```

Comment:
It's nice to be super explicit about what exactly is being transmuted from what to what to make bugs and intent more obvious. I'd prefer all `transmute` being `transmute::<A,B>` with explicit types whenever possible.